src/workerinfo.js

```diff
@@ -174,7 +175,7 @@ class WorkerInfo {
       workerId,
     }, true);
 
-    if (!tasks.length || !worker) {
+    if (!tasks.length || !worker || worker.disabled) {
```

Comment:
It would be a bug if taskSeen was called with a disabled worker -- but we should probably record the taskId in that case.  Otherwise we'd be hiding a bug.

Reply:
Hm, not sure I follow. Presently, a worker can be disabled as long as the worker has not expired. Once expired, a new one will be created. If one wants to make sure a worker doesn't expire, then they could simply change the expiration date of the worker through `declareWorker`. That said, why would it be a bug if `taskSeen` was called with a disabled worker? One of the first thing this method does is check if the worker is disabled. If it is, then it returns without recording the task ID as seen.

Reply:
It would be a bug because it would mean that a task was assigned to a disabled worker..